src/Shape/Box.js

```diff
@@ -107,56 +112,51 @@ export default class Box extends BaseClass {
     // Draw box.
     new Rect()
       .data(filteredData)
-      .x(filteredData[0].x)
-      .y(filteredData[0].y)
-      .config(this._rectConfig)
-      .select(elem("g.d3plus-box", {
-        parent: this._select
-      }).node())
+      .x(this._x)
+      .y(this._y)
+      .select(elem("g.d3plus-Box", {parent: this._select}).node())
+      .config(configPrep.bind(this)(this._rectConfig, "shape"))
       .render();
 
-    // Compute coordinates for median line.
-    const boxTopY = this._y() - this._rectHeight(filteredData[0]) / 2;
-    const medianPoint1X = this._x() - this._rectWidth() / 2;
-    const medianPoint1Y = boxTopY + filteredData[0].third - filteredData[0].median;
-    const medianPoint2X = medianPoint1X + this._rectWidth();
-    const medianPoint2Y = medianPoint1Y;
-
-    const medianData = [
-      {x: medianPoint1X, y: medianPoint1Y},
-      {x: medianPoint2X, y: medianPoint2Y}
-    ];
+    // Construct median data.
+    const medianData = [];
+    filteredData.forEach((d, i) => {
+      const x = this._x(d, i);
+      const y = this._y(d, i) - d.height / 2 + d.third - d.median;
+      medianData.push({x, y, width: this._rectWidth(d, i)});
+    });
 
-    // Draw median line inside the box.
-    new Line()
+    // Draw median.
+    new Rect()
       .data(medianData)
-      .select(elem("g.d3plus-box-median", {
-        parent: this._select
-      }).node())
+      .select(elem("g.d3plus-Box-Median", {parent: this._select}).node())
+      .config(configPrep.bind(this)(this._medianConfig, "shape"))
       .render();
 
     // Draw 2 lines using Whisker class.
-    // Note that this._x() and this._y() are coordinates of center of the rectangle.
-
-    // Construct coordinates for top whisker startpoint.
-    const point1X = this._x();
-    const point1Y = this._y() - this._rectHeight(filteredData[0]) / 2;
-
-    // Construct coordinates for bottom whisker startpoint.
-    const point2X = this._x();
-    const point2Y = this._y() + this._rectHeight(filteredData[0]) / 2;
-
-    const whiskerData = [
-      {x: point1X, y: point1Y, length: this._topWhiskerLength, orient: "top"},
-      {x: point2X, y: point2Y, length: this._bottomWhiskerLength, orient: "bottom"}
-    ];
+    // Construct coordinates for whisker startpoints and push it to the whiskerData.
+    const whiskerData = [];
+    filteredData.forEach((d, i) => { 
+
+      const x = this._x(d, d);
+      const topY = this._y(d, i) - d.height / 2;
+      const bottomY = this._y(d, i) + d.height / 2;
+      const topLength = d.top - d.third;
+      const bottomLength = d.first - d.bottom;
+
+      whiskerData.push(
+        {x, y: topY, length: topLength, orient: "top"},
+        {x, y: bottomY, length: bottomLength, orient: "bottom"}
+      );
+    });
```

Comment:
The same comment from the medianData above applies to the whiskerData: we need to preserve the user's data and index inside of this data array in order to correctly draw the whiskers (the keys `__d3plus__`, `data`, and `i` need to be present in these new objects).

For example, I should be able to change the whisker endpoints like this:
```js
var data = [
  {id: "alpha", value: 200},
  {id: "alpha", value: 20},
  {id: "alpha", value: 180},
  {id: "alpha", value: 40},
  {id: "alpha", value: 170},
  {id: "alpha", value: 125},
  {id: "alpha", value: 74},
  {id: "alpha", value: 80},
  {id: "beta", value: 180},
  {id: "beta", value: 30},
  {id: "beta", value: 120},
  {id: "beta", value: 50},
  {id: "beta", value: 140},
  {id: "beta", value: 115},
  {id: "beta", value: 14},
  {id: "beta", value: 30}
];

var area = new d3plus.Box()
  .data(data)
  .x(d => d.id === "alpha" ? 100 : 200)
  .whiskerConfig({
    endpoint: d => d.id === "alpha" ? "Rect" : "Circle"
  })
  .render();
```